test/common/network/connection_impl_test.cc

```diff
@@ -3086,8 +3093,23 @@ TEST_F(InternalClientConnectionImplTest,
       "");
 }
 
-class ClientConnectionWithCustomRawBufferSocketTest : public ConnectionImplTest {
+class ClientConnectionWithCustomRawBufferSocketTest : public ConnectionImplTestBase,
+                                                      public testing::TestWithParam<Address::Type> {
 protected:
+  void setUpBasicConnection() {
+    Address::InstanceConstSharedPtr address;
+    switch (GetParam()) {
+    case Address::Type::Pipe:
```

Comment:
Thanks for adding the test!
I am a little bit confused why adding the test to cover the address Type::Pipe can help us verify the filter state sharing across downstream filter<->upstream filter<->next downstream filter.

Reply:
The error was that sharing to the upstream connection was guarded by "IP" type, so both pipes and internal addresses were broken. I can't use internal address here since this is a core, and the internal address factory is an extension.